### PR TITLE
Updated AppAdapter.java

### PR DESCRIPTION
**Pull request by TLGINO**
Added function to filter displayed apps in AppAdapter.java in order to remove all system apps (as system apps tend to not contain ads).

### CODE REVIEW

**Review by NikLeberg — COMMENTED** [review]
Hi @TLGINO Thank you for your contribution!
Quite unexpected to have an open PR this fast to be honest. Are you actively using our app?
I agree that system apps don't have ads and your change would clean up the list quite a bit. I'll review it and maybe tweak the code formatting a little bit.

File: app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java [hunk 1]
```diff
@@ -28,7 +29,7 @@
  * - https://developer.android.com/training/package-visibility/declaring
  * - https://stackoverflow.com/questions/60679685/what-does-query-all-packages-permission-do
  *
- * @author Niklaus Leuenberger
+ * @author Niklaus Leuenberger TLGINO
```
Comment (NikLeberg) [inline app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java:32]:
```suggestion
 * @author Niklaus Leuenberger, TLGINO
```

File: app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java [hunk 2]
```diff
@@ -47,12 +48,28 @@ public AppAdapter(@NonNull Context context) {
...
+        String[] packageNames = sortPackages(packages);
         super.addAll(packageNames);
         // Loading all package icons upfront would require too much processing time on the main
         // ui tread and create animation stutters. So this is handled in getView().
     }
 
+    /**
+     * Returns a list of names corresponding to the non-system file apps
+     * @param packages list of found packages
+     * @return String[] of package names
```
Comment (NikLeberg) [inline app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java:60]:
```suggestion
     * Returns a list of names corresponding to the non-system file apps.
     *
     * @param packages List of found packages.
     * @return String[] Array of package names.
```

File: app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java [hunk 3]
```diff
@@ -47,12 +48,28 @@ public AppAdapter(@NonNull Context context) {
...
+     */
+    public String[] sortPackages(List<ApplicationInfo> packages) {
+        ArrayList<String> names = new ArrayList<>();
+        for (ApplicationInfo applicationInfo : packages){
+            if ((applicationInfo.flags & ApplicationInfo.FLAG_SYSTEM) == 0){
+                names.add(applicationInfo.packageName);
+            }
+        }
+        return names.stream().toArray(String[]::new);
```
Comment (NikLeberg) [inline app/src/main/java/ch/bfh/adaid/gui/rule/AppAdapter.java:69]:
```suggestion
        return names.toArray(new String[0]);
```

**Review by NikLeberg — CHANGES_REQUESTED** [review]
Hi @TLGINO
I redid the patch as review suggestions. Could you apply them please?

**Comment by TLGINO** [comment]
Looks good!

**Review by NikLeberg — APPROVED** [review]

**Comment by NikLeberg** [comment]
Thanks!

**Pull request merged**